src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -399,9 +399,26 @@ class CharacterPrediction implements IPrediction {
 		return cursor.clone().moveTo(this.appliedAt.pos) + input;
 	}
 
-	public matches(input: StringReader, lookBehind?: IPrediction) {
+	public matches(input: StringReader, lookBehind?: IPrediction, cursor?: Cursor) {
 		let startIndex = input.index;
 
+		const cursorMove = input.eatRe(/^\x1b\[(\d+);(\d+)H/);
+		if (cursorMove) {
+			const x = this.appliedAt!.pos.x;
+			// We need to follow the cursor movements
+			cursor!.moveTo({
```

Comment:
note that the cursor is mutable, and you're actually moving it here. You probably want to `clone()` the cursor before doing these checks to avoid throwing off state (particularly if the character doesn't actually match).